hamt/hamt.go

```diff
@@ -229,6 +229,26 @@ func (ds *Shard) Set(ctx context.Context, name string, nd ipld.Node) error {
 	return err
 }
 
+// Set sets 'name' = nd in the HAMT, using directly the information in the
+// given link. This avoids writing the given node, then reading it to making a
+// link out of it.
+func (ds *Shard) SetLink(ctx context.Context, name string, lnk *ipld.Link) error {
+	hv := newHashBits(name)
```

Comment:
Not required, but if in the future someone wants to extract this into reusable `ds.SwapLink` function for similar types of optimizations they can do so. Although it's really just a few copied lines of code.